SoftLayer/managers/storage_utils.py

```diff
@@ -942,13 +942,14 @@ def prepare_duplicate_order_object(manager, origin_volume, iops, tier,
     # Determine the IOPS or tier level for the duplicate volume, along with
     # the type and prices for the order
     origin_storage_type = origin_volume['storageType']['keyName']
-    if origin_storage_type == 'PERFORMANCE_BLOCK_STORAGE'\
-            or origin_storage_type == 'PERFORMANCE_BLOCK_STORAGE_REPLICANT'\
-            or origin_storage_type == 'PERFORMANCE_FILE_STORAGE'\
-            or origin_storage_type == 'PERFORMANCE_FILE_STORAGE_REPLICANT':
+    if 'PERFORMANCE' in origin_storage_type:
         volume_is_performance = True
-        iops = _validate_dupl_performance_iops(
-            origin_volume, iops, duplicate_size)
+        if iops is None:
+            if isinstance(utils.lookup(origin_volume, 'provisionedIops'), str):
```

Comment:
I think this check is a little unneeded. What if the API changes provisionedIops to be an int in the future?
I think just doing ```iops = int(origin_volume.get('provisionedIops', 0))``` should be sufficient. Then you can make sure that value is > 0 and then throw your exception.

Reply:
I can do that instead, no problem.